src/test/suite/completion.test.ts

```diff
@@ -316,6 +345,38 @@ describe("Should do completion", () => {
       stdinMock.write(new SimpleAutocompleteRequestMatcher("return"), "utf8")
     ).never();
   });
+  it("should render suggestion after debounce", async () => {
```

Comment:
please add a test to the cancellation logic:
1. set long debounce
2. request 2 different completions requests with short sleep between them
3. assert request is fired once (the second request)